spacemeshos/go-spacemesh
# [Merged by Bors] - recover 3of3: add checkpoint systest

## Motivation
part of https://github.com/spacemeshos/go-spacemesh/issues/4090
PR 3 of 3 to break down large PR https://github.com/spacemeshos/go-spacemesh/pull/4387
loosely dependent on PR 1 of 3 https://github.com/spacemeshos/go-spacemesh/pull/4404

## Changes
- change boostrapper to serve checkpoint file

- checkpoint systest does the following:
  - submit transactions to spawn account and transfer coins
  - get accounts nonce/balance
  - issue checkpoint RPC to every node
  - check all the returned checkpoint data are the same
  - issue recovery RPC to every node (node will copy the local checkpoint file to $dataDir/recovery/) and restart
  - check that all accounts have the same nonce/balance as before the checkpoint
  - check all miners start smeshing
  - make boostrapper serve the checkpoint data
  - add two new nodes with --checkpoint_file=[http://bootstrapper-0:80/checkpoint](http://bootstrapper-0/checkpoint) and --restore-layer
  - check all miners are smeshing
